spec/Section 7 -- Response.md

```diff
@@ -9,40 +9,59 @@ the case that any _execution error_ was raised and replaced with {null}.
 
 ## Response Format
 
-A GraphQL request returns either a _response_ or a _response stream_.
+:: A GraphQL request returns a _response_. A _response_ is either an _execution
+result_, a _request error result_, or a _response stream_.
 
-### Response
+### Execution Result
 
-:: A GraphQL request returns a _response_ when the GraphQL operation is a query
-or mutation. A _response_ must be a map.
+:: A GraphQL request returns an _execution result_ when the GraphQL operation is
+a query or mutation and the request included execution. Additionally, an
+execution result is the event emitted by a _response stream_.
 
-If the request raised any errors, the response map must contain an entry with
-key {"errors"}. The value of this entry is described in the "Errors" section. If
-the request completed without raising any errors, this entry must not be
+An _execution result_ must be map.
+
+The _execution result_ must contain an entry with key {"data"}. The value of
+this entry is described in the "Data" section.
+
+If execution raised any errors, the _execution result_ must contain an entry
+with key {"errors"}. The value of this entry must be a non-empty list of
+_execution error_ raised during execution. Each error must be a map as described
+in the "Errors" section below. If the request completed without raising any
+errors, this entry must not be present.
```

Comment:
I moved and adjusted some of the text from the Errors section. Placing it under `Execution Result` and `Request Error Result` so we can describe the specific behavior (must be included in  `Request Error Result`, may be included in `Execution Result`) without the Errors section having conditions for each type.